### PR TITLE
Pinecone fixes

### PR DESCRIPTION
**Pull request by KE7**

### CODE REVIEW

**Comment by KE7** [comment]
Changes based on testing where DSPy was used for a different project: https://github.com/KE7/scenicNL/blob/main/notebooks/dspy.ipynb 

@okhat please send me an email (k.e@berk) if you want access to that repo. It's for a project I'm working on with Matei (he already has access)

File: dspy/retrieve/pinecone_rm.py [hunk 1]
```diff
@@ -3,6 +3,9 @@
 Author: Dhar Rawal (@drawal1)
 """
 
+import torch
```
Comment (okhat) [inline dspy/retrieve/pinecone_rm.py:6]: We can't import this here, torch isn't needed for other stuff. This import should be deferred, it should be inside a function call for instance.

File: dspy/retrieve/pinecone_rm.py [hunk 2]
```diff
@@ -3,6 +3,9 @@
...
+from dsp.utils import dotdict
+from transformers import AutoTokenizer, AutoModel
```
Comment (okhat) [inline dspy/retrieve/pinecone_rm.py:8]: Same here.

**Comment by okhat** [comment]
Thank you! Left two small comments.

**Comment by okhat** [comment]
@KE7 Following up after NeurIPS in case you have a chance to tweak these two issues above!

**Comment by KE7** [comment]
@okhat Should be all good now :)

**Pull request merged**